_docs/rfc/imports.md

```diff
@@ -0,0 +1,963 @@
+
+If you wanted to be explicit and avoid the implicit `merge` of the blocks, you can also explicitly set the attributes of
+the block in the child config:
+
+```hcl
+import "root" {
+  config_path = find_in_parent_folders("root.hcl")
+}
+
+remote_state {
+  backend = import.root.remote_state.backend
+  config  = merge(
+    import.root.remote_state.config,
+    {
+      # The relative path between the root terragrunt directory and the current terragrunt file directory.
+      key =  relpath(import.root.terragrunt_dir, get_terragrunt_dir())
+    },
+  )
+}
+```
+
+NOTE: `relpath` does not currently exist, and must be implemented as a terragrunt helper function.
+
+The advantage of this configuration is that we're only pulling the `remote_state` configuration to the child. The other
+blocks and attributes (e.g `inputs`) is not inherited to the child config. This was something that was not possible to
+do with `include`.
+
+#### Why can't we use the merge function?
+
+In the above example, we have to explicitly declare the block. A natural extension of this pattern is to use the `merge`
+function directly on the `remote_state`:
+
+```hcl
+remote_state = merge(
+  import.root.remote_state,
+  {
+    config = {
+      key = relpath(import.root.terragrunt_dir, get_terragrunt_dir())
+    }
+  }
+)
+```
+
+Unfortunately, this is a syntax error because `remote_state` is a block, and not an attribute. You can not directly set
+blocks in this way.
+
+
+
+
+## Alternatives
+
+### In consideration
+
+#### Single terragrunt.hcl file per environment
+
+This implementation introduces a new block `module` that replaces `dependency`, `terraform`, and `inputs`. This approach
+is documented in [#759](https://github.com/gruntwork-io/terragrunt/issues/759).
+
+In addition to the general analysis of that proposal, here are the list of Pros and Cons related to the problem of
+sharing config:
+
+Pros:
+
+- Circumvent the reference problem by having all the references in a single shared scope.
+
+Cons:
+
+- If you need to share configurations across environments, you would still need one of the alternative approaches to
+  import the other config. This, however, is likely to be a rare occurrence.
+- This is a drastic change to Terragrunt, completely changing the way it works.
+
+
+Let's walk through how each of the import use cases look like with this implementation:
+
+**Hierarchical variables**
+
+In this approach, a hierarchy of variables is unnecessary because all the blocks are defined in a single scope. However,
+depending on the scope of `locals`, certain things like reusing a repetitive variable becomes more challenging.
+
+To understand this, consider a use case where you are operating in two regions, `us-east-1` and `eu-west-1`. To simplify
+this example, consider a limited application deployment where you have two modules: `vpc` and `app`.
+
+If we assume that `locals` are scoped within the file that they are defined, then you can implement this by separating
```

Comment:
Why would locals be scoped to a file? The idea with the new proposal is that all `.hcl` files in a single folder are processed as if they are in a single file, just as Terraform processes all `.tf` files in a single folder as if they are in a single file.

Reply:
There is an advantage to having vars scoped to the file so that you don't have to keep picking new names for a thing like `region`.

Reply:
Agreed, but I think in net, treating all `.hcl` files in a folder as if they were in one `.hcl` file, just as Terraform does with `.tf` files, is the better trade-off.

Reply:
I'll update to note that it is preferred to have scope of locals shared, but mention the possibility of having file scoped vars, its advantages and disadvantages.